sql/core/src/test/resources/sql-tests/inputs/cte.sql

```diff
@@ -24,6 +24,10 @@ SELECT t1.id AS c1,
 FROM   CTE1 t1
        CROSS JOIN CTE1 t2;
 
+-- CTE with column alias
+WITH t(x) AS (SELECT 1)
```

Comment:
Could you add more test cases? For example, can a with clause in a subquery shadow a with clause in an enclosing query with the same name? Another example, use with clauses in a subquery expression?

Reply:
I wanted to focus on column aliases in this PR, but here is another open here which focuses on nested `WITH` clauses: https://github.com/apache/spark/pull/24831

Reply:
Yep. For the nested `WITH`, #24831 would be a better place to add them.
After merging this, let's proceed as @gatorsmile suggested, @peter-toth .